nlp_utilities_test.jl

```diff
@@ -0,0 +1,121 @@
+using JuMP
+using Ipopt
+using Test
+"""
+    create_nonlinear_jump_model()
+
+Create a nonlinear jump model from the example in:
+JuMP Tutorial for Querying Hessians:
+https://github.com/jump-dev/JuMP.jl/blob/301d46e81cb66c74c6e22cd89fb89ced740f157b/docs/src/tutorials/nonlinear/querying_hessians.jl#L67-L72
+"""
+function create_nonlinear_jump_model()
+    model = Model(Ipopt.Optimizer)
+    set_silent(model)
+    @variable(model, p ∈ MOI.Parameter(1.0))
+    @variable(model, p2 ∈ MOI.Parameter(2.0))
+    @variable(model, p3 ∈ MOI.Parameter(100.0))
```

Comment:
I'm not that familiar with `variable in Parameter` declarations. This seems similar to declaring the variable then calling `fix(p, value)`, with the following differences:
- We use `FixRef` and `is_fixed` for the fixed variable and `ParameterRef` and `is_parameter` for the parameter variable
- The dual can be obtained for a `FixRef` constraint, but not for a `ParameterRef` constraint.
- `set_parameter_value` and `parameter_value` can be used to set and get a parameter's value, regardless of whether the model has been modified since a solve.

Any other differences I should be aware of?

Reply:
@joaquimg